crates/core/transaction/src/plan/build.rs

```diff
@@ -14,72 +17,35 @@ use crate::{
 };
 
 impl TransactionPlan {
-    /// Build the transaction this plan describes.
-    ///
-    /// To turn a transaction plan into an unauthorized transaction, we need:
-    ///
-    /// - `fvk`, the [`FullViewingKey`] for the source funds;
-    /// - `witness_data`, the [`WitnessData`] used for proving;
-    ///
-    pub fn build(
+    /// Builds a [`TransactionPlan`] by slotting in the
+    /// provided prebuilt actions instead of using the
+    /// [`ActionPlan`]s in the TransactionPlan.
+    /// Arguments:
+    ///     self: `TransactionPlan`
+    ///     fvk: `FullViewingKey`
+    ///     witness_data: `WitnessData`
+    /// Returns: `Transaction`
+    pub fn build_unauth_with_actions(
         self,
-        fvk: &FullViewingKey,
+        mut actions: Vec<Action>,
+        fvk: FullViewingKey,
         witness_data: WitnessData,
-    ) -> Result<UnauthTransaction> {
-        let mut actions = Vec::new();
-        let mut synthetic_blinding_factor = Fr::zero();
-
+    ) -> Result<Transaction> {
         // Add the memo.
         let mut memo: Option<MemoCiphertext> = None;
-        let mut memo_key: Option<PayloadKey> = None;
+        let mut _memo_key: Option<PayloadKey> = None;
```

Comment:
What happened to the memo key?

Reply:
In an earlier comment in this PR, I mentioned: **"I'm trying to think through the best way to handle the memo field. There's also code duplication happening between the [build_action](https://github.com/penumbra-zone/penumbra/blob/tx-decompose/crates/wasm/src/wasm_planner.rs#L98), [build_unauth_with_actions](https://github.com/penumbra-zone/penumbra/blob/tx-decompose/crates/core/transaction/src/plan/build.rs#L34) and [build](https://github.com/penumbra-zone/penumbra/blob/tx-decompose/crates/core/transaction/src/plan/build.rs#L258) methods with respect to the memo field that need to be addressed."**. 

The way we've constructed this, the `memo_key` field isn't required in the computation of [build_unauth_with_actions](https://github.com/penumbra-zone/penumbra/blob/tx-decompose/crates/core/transaction/src/plan/build.rs#L28), and the `memo` field isn't required in the computation of [build](https://github.com/penumbra-zone/penumbra/blob/tx-decompose/crates/core/transaction/src/plan/build.rs#L261) and [build_concurrent](https://github.com/penumbra-zone/penumbra/blob/tx-decompose/crates/core/transaction/src/plan/build.rs#L321). The [build_actions](https://github.com/penumbra-zone/penumbra/blob/tx-decompose/crates/wasm/src/wasm_planner.rs#L97) method in the wasm crate also requires the memo field. 

Do you have any suggestions on how to unify the memo field across these different implementations to reduce code duplication?